### ISSUE TITLE
[bug]: A disabled product will crash the PDP

### ISSUE THREAD

**Issue by Jordaneisenburger** [magento/pwa-studio#2627]
**Describe the bug**

**To reproduce**
1. Go to a product page
2. disable the product in Magento
3. reindex & flush cache
4. force reload the PDP

**Expected behavior**
Show a 404

**Screenshots**
![Screenshot 2020-08-14 at 10 20 28](https://user-images.githubusercontent.com/19858728/90229197-cc371f00-de17-11ea-9d6a-e990424b3722.png)

**Possible solutions**
In peregrine the useProduct.js hook that does the 'network-only' request will always return `data` and therefore it gets into the if statement. Quick fix would be to change the statement to `if(data?.productDetail.items[0])`. This will make the product return `null` and shows the product is currently out of stock.

But we don't want to show a out of stock message but rather show a 404 since it's disabled.

**Please complete the following device information:**
 - Device [e.g. iPhone6, PC, Mac, Pixel3]:
 - OS [e.g. iOS8.1, Windows 10]:
 - Browser [e.g. Chrome, Safari]:
 - Browser Version [e.g. 22]:
 - Magento Version:
 - PWA Studio Version:
 - NPM version `npm -v`:
 - Node Version `node -v`:

**Please let us know what packages this bug is in regards to:**
- [ ] `venia-concept`
- [x] `venia-ui`
- [ ] `pwa-buildpack`
- [x] `peregrine`
- [ ] `pwa-devdocs`
- [ ] `upward-js`
- [ ] `upward-spec`
- [ ] `create-pwa`

**Comment by sirugh** [comment]
> But we don't want to show a out of stock message but rather show a 404 since it's disabled.

Have we solved the disabled product ui before? My assumption was that we just display it as out of stock right now.

**Comment by Jordaneisenburger** [comment]
Yeah we've had the same issue for categories when no products are found. Not sure what UI is shown when a product is out of stock but we should redirect to a 404 when the `data?.productDetail.items[0]` array is empty right ?

**Comment by huykon** [comment]
I have seen if product loaded on PWA then disable product on Magento backend then reload product still have enable status without re-call GraphQl API.

**Comment by awilcoxa** [comment]
Created PWA-865

**Comment by NandiniJivainfotech123** [comment]
@magento I am working on this

**Comment by sirugh** [comment]
@NandiniJivainfotech123 thanks for picking this up - we're not sure that this is still a problem so it would be good to first check the repro steps.

If you look at [the code](https://github.com/magento/pwa-studio/blob/develop/packages/peregrine/lib/talons/RootComponents/Product/useProduct.js#L34-L36) that is throwing the error, you'll see that we should never end up in `mapProduct` if `items[0]` is falsy.

Please let us know if you're still able to repro the issue!

**Comment by NandiniJivainfotech123** [comment]
I am not able to repro the issue

**Comment by sirugh** [comment]
Closing as previously fixed :)
